From f004aa1dec6e2e206be025de15b115d60f2b21e3 Mon Sep 17 00:00:00 2001
From: Mathieu Olivari <mathieu@codeaurora.org>
Date: Tue, 21 Apr 2015 19:09:07 -0700
Subject: [PATCH 9/9] ARM: qcom: automatically select PCI_DOMAINS if PCI is
 enabled

If multiple PCIe devices are present in the system, the kernel will
panic at boot time when trying to scan the PCI buses. This happens on
IPQ806x based platforms, which has 3 PCIe ports.

Enabling this option allows the kernel to assign the pci-domains
according to the device-tree content. This allows multiple PCIe
controllers to coexist in the system.

Signed-off-by: Mathieu Olivari <mathieu@codeaurora.org>
---
 arch/arm/mach-qcom/Kconfig | 1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm/mach-qcom/Kconfig
+++ b/arch/arm/mach-qcom/Kconfig
@@ -5,6 +5,7 @@ menuconfig ARCH_QCOM
 	select ARM_AMBA
 	select PINCTRL
 	select QCOM_SCM if SMP
+	select PCI_DOMAINS if PCI
 	help
 	  Support for Qualcomm's devicetree based systems.
 
